llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -24228,8 +24228,13 @@ SDValue X86TargetLowering::LowerSETCC(SDValue Op, SelectionDAG &DAG) const {
   if (Subtarget.hasAVX10_2()) {
     if (CC == ISD::SETOEQ || CC == ISD::SETUNE) {
       auto NewCC = (CC == ISD::SETOEQ) ? X86::COND_E : (X86::COND_NE);
-      return getSETCC(NewCC, DAG.getNode(X86ISD::UCOMX, dl, MVT::i32, Op0, Op1),
-                      dl, DAG);
+      auto isValidType = [&](MVT Type) {
+        return Type == MVT::f16 || Type == MVT::f32 || Type == MVT::f64;
```

Comment:
What's the root cause for this? My understanding is #113567 only set these 3 types `Custom`, how can we reach with f80? Is f80 special? I checked bf16 doesn't crash even without this change.

Reply:
There are 3 checks above this one

`if (isSoftF16(Op0.getValueType(), Subtarget))`, 
`if (Op0.getValueType() == MVT::f128)`, 
`if (Op0.getSimpleValueType().isInteger())`

So, it seems `f80` is the only type that can reach it. I will simplify the check for just this type.

Reply:
`isSoftF16` return false for bf16 under AVX10.2. Can you investigate it a bit more?

Reply:
I wasn't able to create an example with `bf16` that reaches this code, in my example I see `bfloat` bitcasted to a `i16` and then promoted to `f32` during legalization.
```
Legalizing node: t4: f16,ch = CopyFromReg t0, Register:f16 %1
Legalizing node: t28: i16 = bitcast t4
Legalizing node: t31: f32 = bf16_to_fp t28
```

Reply:
I see, thanks! I assumed bf16 type is legal with AVX10.2 but it's not. Maybe adding an assert for bf16.